hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -2058,7 +2080,7 @@ BatchedListEntries<OpenFileEntry> listOpenFiles(long prevId,
 
   public BatchedListEntries<OpenFileEntry> getFilesBlockingDecom(long prevId,
       String path) {
-    assert hasReadLock();
+    assert hasReadLock(RwLockMode.FS);
```

Comment:
I think we should use GLOBAL mode here, because below codes contain iterate `blockManager.getDatanodeManager().getDatanodes()`

Reply:
Thanks @hfutatzhanghb for your review. 
Both `dataNode.getLeavingServiceStatus().getOpenFiles()` and `blockManager.getDatanodeManager().getDatanodes())` are already thread-safe, so the BMLock is unnecessary here.